gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -306,14 +320,18 @@ protected void setTaskStagingAndOutputDirs() {
    * plus {@link #TASK_STAGING_DIR_NAME}.
    */
   private void setTaskStagingDir() {
+    String workingDir;
     if (this.jobState.contains(ConfigurationKeys.WRITER_STAGING_DIR)) {
```

Comment:
This condition should be a NOOP. This method only takes effect when `WRITER_STAGING_DIR` is not specified.